### PR TITLE
fix: make l1_fee_scalar f64

### PR DESCRIPTION
**Pull request by 0xZerohero**
## Motivation
Related to https://github.com/paradigmxyz/reth/issues/6547
In op-geth l1FeeScalar is represented as a float and is serialized to string
```go
FeeScalar  *big.Float `json:"l1FeeScalar,omitempty"` // always nil after Ecotone hardfork
```
```go
if receipt.FeeScalar != nil { // removed in Ecotone
	fields["l1FeeScalar"] = receipt.FeeScalar.String()
}
```

This introduces a breaking change in reth and need to be synchronized with https://github.com/paradigmxyz/reth/pull/6575

## PR Checklist

- [x] Added Tests
- [ ] Added Documentation
- [x] Breaking changes

### CODE REVIEW

File: crates/rpc-types/src/eth/transaction/optimism.rs [hunk 1]
```diff
@@ -30,8 +30,8 @@ pub struct OptimismTransactionReceiptFields {
     #[serde(skip_serializing_if = "Option::is_none")]
     pub l1_fee: Option<U256>,
     /// L1 fee scalar for the transaction
-    #[serde(skip_serializing_if = "Option::is_none")]
-    pub l1_fee_scalar: Option<U256>,
+    #[serde(default, skip_serializing_if = "Option::is_none", with = "l1_fee_scalar_serde")]
```
Comment (mattsse) [inline crates/rpc-types/src/eth/transaction/optimism.rs:33]:
```suggestion
    #[serde(skip_serializing_if = "Option::is_none", with = "l1_fee_scalar_serde")]
```

**Review by mattsse — APPROVED** [review]
lgtm, 1 nit

File: crates/rpc-types/src/eth/transaction/optimism.rs (hunk 1 above)
Comment (0xZerohero) [reply crates/rpc-types/src/eth/transaction/optimism.rs:33]: Without the serde(default) the deserialization works when the field is Value::Null, but fails when it's not set.
https://stackoverflow.com/a/44303505

```
thread 'eth::transaction::optimism::tests::deserialize_l1_fee_scalar' panicked at crates/rpc-types/src/eth/transaction/optimism.rs:126:88:
called `Result::unwrap()` on an `Err` value: Error("missing field `l1FeeScalar`", line: 0, column: 0)

```

Should I remove it ?

Comment (mattsse) [reply crates/rpc-types/src/eth/transaction/optimism.rs:33]: TIL

**Pull request merged**